docs/topics/dokka-migration.md

```diff
@@ -77,157 +95,423 @@ After updating the Dokka Gradle plugin to v2, follow the migration steps applica
+
+```kotlin
+
+// CustomPlugin.kt
+
+class CustomPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.extensions.configure<DokkaExtension> {
+
+            dokkaPublications.named("html") {
+                suppressInheritedMembers.set(true)
+                failOnWarning.set(true)
+            }
+
+            dokkaSourceSets.named("main") {
+                includes.from("README.md")
+                sourceLink {
+                    localDirectory.set(project.file("src/main/kotlin"))
+                    remoteUrl.set(URI("https://example.com/src"))
+                    remoteLineSuffix.set("#L")
+                }
+            }
+
+            pluginsConfiguration.named<org.jetbrains.dokka.gradle.engine.plugins.DokkaHtmlPluginParameters>("pluginsConfiguration") {
+                customStyleSheets.from("styles.css")
+                customAssets.from("logo.png")
+                footerMessage.set("(c) Your Company")
+            }
+        }
+    }
+}
+```
+
+</tab>
+</tabs>
+
+#### Visibility settings
+
+The `documentedVisibilities` property has changed from `Visibility.PUBLIC` to `VisibilityModifier.Public`.
+
+Previous configuration:
+
+```kotlin
+import org.jetbrains.dokka.DokkaConfiguration.Visibility
+
+// ...
+documentedVisibilities.set(
+    setOf(Visibility.PUBLIC)
+) 
+```
+
+New configuration:
+
+```kotlin
+import org.jetbrains.dokka.gradle.engine.parameters.VisibilityModifier
+
+// ...
+documentedVisibilities.set(
+    setOf(VisibilityModifier.Public)
+)
+
+// OR
+
+documentedVisibilities(VisibilityModifier.Public)
+```
+
+Additionally, DGP v2 has a [utility function](https://github.com/Kotlin/dokka/blob/220922378e6c68eb148fda2ec80528a1b81478c9/dokka-runners/dokka-gradle-plugin/src/main/kotlin/engine/parameters/HasConfigurableVisibilityModifiers.kt#L14-L16) for adding documented visibilities:
+
+```kotlin
+fun documentedVisibilities(vararg visibilities: VisibilityModifier): Unit =
+    documentedVisibilities.set(visibilities.asList()) 
+```
+
+#### Source links
+
+Allow users to navigate from the generated documentation to the corresponding source code in a remote repository. 
+To configure source links, use the `dokkaSourceSets.main{}` block.
+
+Previous configuration:
+
+```kotlin
+    tasks.withType<DokkaTask>().configureEach {
+    dokkaSourceSets {
+        named("main") {
+            sourceLink {
+                localDirectory.set(file("src/main/kotlin"))
+                remoteUrl.set(URL("https://github.com/your-repo"))
+                remoteLineSuffix.set("#L")
+            }
+        }
+    }
+}
+```
+
+New configuration:
+
+Using `build.gradle.kts` files differs from using `.kt` files (such as custom plugins)
+due to the use of type-safe accessors in Kotlin DSL.
+
+<tabs group="dokka-configuration">
+<tab title="Gradle configuration file" group-key="gradle">
+
+```kotlin
+
+// build.gradle.kts
+
+dokka {
+    moduleName.set("Project Name")
```

Comment:
probably we don't need `moduleName` and `includes` in this sample (and in `CustomPlugin.kt`) as they are not present in `old configuration`